specification/cognitiveservices/data-plane/EntitySearch/v1.0/EntitySearch.json

```diff
@@ -0,0 +1,935 @@
+            "required": false,
+            "type": "string"
+          },
+          {
+            "name": "cc",
+            "in": "query",
+            "description": "A 2-character country code of the country where the results come from. This API supports only the United States market. If you specify this query parameter, it must be set to us. If you set this parameter, you must also specify the Accept-Language header. Bing uses the first supported language it finds from the languages list, and combine that language with the country code that you specify to determine the market to return results for. If the languages list does not include a supported language, Bing finds the closest language and market that supports the request, or it may use an aggregated or default market for the results instead of a specified one. You should use this query parameter and the Accept-Language query parameter only if you specify multiple languages; otherwise, you should use the mkt and setLang query parameters. This parameter and the mkt query parameter are mutually exclusive—do not specify both.",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "name": "mkt",
+            "in": "query",
+            "description": "The market where the results come from. You are strongly encouraged to always specify the market, if known. Specifying the market helps Bing route the request and return an appropriate and optimal response. This parameter and the cc query parameter are mutually exclusive—do not specify both.",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "name": "q",
+            "in": "query",
+            "description": "The user's search term.",
+            "required": false,
```

Comment:
Just out of curiosity: How is this not required?

Reply:
Actually a really good point. Technically the API 'works' without it it just doesn't do anything 😜

Reply:
lol - I'd definitely mark it as required then. I think it makes sense to model the API as what it is *useful* for rather than what it technically accepts, that makes for "useful" rather than "technically correct" SDKs 😆